minichain/persistence.py

```
@@ -261,62 +265,59 @@ def _load_snapshot_from_sqlite(db_path: str) -> dict[str, Any]:
 
-import time
 
 def _ensure_banned_peers_table(conn: sqlite3.Connection) -> None:
     conn.execute(
         "CREATE TABLE IF NOT EXISTS banned_peers (peer_id TEXT PRIMARY KEY, reason TEXT, timestamp REAL)"
     )
 
-def ban_peer(peer_id: str, reason: str, path: str = ".") -> None:
+
+@contextmanager
+def _banned_peers_conn(path: str, create: bool):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_banned_peers_conn`

(ANN202)